src/Compilers/CSharp/Portable/Symbols/Source/SourceDelegateClonedParameterSymbolForBeginAndEndInvoke.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SourceDelegateClonedParameterSymbolForBeginAndEndInvoke : SourceClonedParameterSymbol
+    {
+        internal SourceDelegateClonedParameterSymbolForBeginAndEndInvoke(SourceParameterSymbol originalParam, Symbol newOwner, int newOrdinal)
+            : base(originalParam, newOwner, newOrdinal, suppressOptional: true)
+        {
```

Comment:
>{ [](http://example.com/codeflow?start=8&length=1)

Do we want to assert that the ```originalParam``` is not optional?

Reply:
@AlekseyTs I think I should be asserting `this.IsOptional`, not `originalParam.IsOptional`.

`originalParam` can be optional, but it's suppressed via `suppressOptional: true`.

I'm currently blocked on adding the assertion. See https://github.com/dotnet/roslyn/pull/54389.

Reply:
> I think I should be asserting this.IsOptional

Agree